src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/client/FeatureServiceClient.scala

```diff
@@ -1,43 +1,59 @@
 package com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client
 
-import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.Geofence
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.{Geofence, Logger}
 import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.dto.{FeatureServiceFeature, FeatureServiceResponse}
 import net.liftweb.json
 
 import scala.io.Source
+import scala.util.{Failure, Success, Try}
 
 @SerialVersionUID(100L)
-class FeatureServiceClient(host: String) extends Serializable {
+class FeatureServiceClient(host: String) extends Serializable with Logger {
   def bbox(geofence: Geofence): Iterable[FeatureServiceFeature] = {
-    parseResponse(fetchBboxResponse(geofence))
+    unpack(fetchBboxResponse(geofence), "bbox")
   }
 
   def point(latitude: Double, longitude: Double): Iterable[FeatureServiceFeature] = {
-    parseResponse(fetchPointResponse(latitude = latitude, longitude = longitude))
+    unpack(fetchPointResponse(latitude = latitude, longitude = longitude), "point")
   }
 
   def name(names: Iterable[String]): Iterable[FeatureServiceFeature] = {
-    parseResponse(fetchNameResponse(names))
+    unpack(fetchNameResponse(names), "name")
   }
 
-  private def parseResponse(response: String): Iterable[FeatureServiceFeature] = {
+  private def unpack(responseBody: Try[String], endpointName: String): Iterable[FeatureServiceFeature] = {
```

Comment:
The error is still swallowed here. I would expect that clients of the public methods of this class should be able to distinguish between empty results and an internal error.